src/lib/mdns/MdnsCache.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <cstdint>
+
+#include <core/CHIPError.h>
+#include <core/PeerId.h>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/InetLayer.h>
+#include <system/SystemTimer.h>
+#include <system/TimeSource.h>
+
+// set MDNS_LOGGING to enable logging -- sometimes used in debug/test programs -- traces the behavior
+#ifdef MDNS_LOGGING
+#define MdnsLogProgress ChipLogProgress
+#else
+#define MdnsLogProgress(...)
+#endif
+
+namespace chip {
+namespace Mdns {
+
+template <size_t CACHE_SIZE>
+class MdnsCache
+{
+public:
+    MdnsCache() : elementsUsed(CACHE_SIZE)
+    {
+        nullPeerId.SetFabricId(kUndefinedFabricId);
```

Comment:
It's a default constructed `PeerId`, so I'm not sure this code and `nullPeerId` is necessary. You can just clear peer IDs by assigning `PeerId{}`.

Reply:
Not sure I follow.    I'm setting nullPeerId as a value to indicate a "free entry".    A flag could be used -- but somehow, entries in the cache have to be assigned "unused" when necessary.

Reply:
I mean that `nullPeerId` == `PeerId{}`, so you don't need to define any new value.

Reply:
ok, so just

PeerId nullPeerId

is sufficient by default....